dynet/expr.h

```diff
@@ -1610,7 +1610,32 @@ Expression kmh_ngram(const Expression& x, unsigned n);
 /**
  * \ingroup convolutionoperations
  * \brief conv2d without bias
- * \details 2D convolution operators without bias parameters.
+ * \details
+ *   2D convolution operator without bias parameters.
+ *   'VALID' and 'SAME' convlutions are supported. Assume:
+ *   - Input feature maps: XH x XW x XC x N
+ *   - Filters: FH x FW x XC x FC 
+ *   - Strides: strides[0] and strides[1] are row (h) and col (w) stride, respectively.
+ *
+ *   For the *SAME* convolution: the output height (YH) and width (YW) are computed as:
+ *   - YH = ceil(float(XH) / float(strides[0]))
+ *   - YW = ceil(float(XW) / float(strides[1]))
+ *   and the paddings are computed as:
+ *   - pad_along_height = max((YH - 1) * strides[0] + FH - XH, 0)
+ *   - pad_along_width = max((YW - 1) * strides[1] + FW - XW, 0)
+ *   - pad_top = pad_along_height / 2
+ *   - pad_bottom = pad_along_height - pad_top
+ *   - pad_left = pad_along_width / 2
+ *   - pad_right = pad_along_width - pad_left
+ *
+ *   For the *VALID* convolution: the output height (YH) and width (YW) are computed as:
+ *   - YH = ceil(float(XH - FH + 1) / float(strides[0]))
+ *   - YW = ceil(float(XW - FW + 1) / float(strides[1]))
+ *   and the paddings are always zeros.
+ *
+ *   Think about when stride is 1, the distinction:
```

Comment:
Maybe move this to the front of the explanation, then describe the details later?